lib/compress/index.js

```diff
@@ -1598,6 +1598,38 @@ def_optimize(AST_Switch, function(self, compressor) {
     while (i < len) eliminate_branch(self.body[i++], body[body.length - 1]);
     self.body = body;
 
+    // group equivalent branches so they will be located next to each other,
+    // that way the next micro-optimization will merge them.
+    // ** bail micro-optimization if not a simple switch case with breaks
+    if (body.every((branch, i) =>
+            (body.length - 1) === i // last branch
+            || branch instanceof AST_Default // or default:
+            || !branch.expression.has_side_effects(compressor) // or no_side_effects
+            && (
+                branch.body.length === 0 // or empty
+                || is_break(branch.body[branch.body.length - 1], compressor) // or has a break
+                || is_return(branch.body[branch.body.length - 1]) // or has a return
```

Comment:
It's not clear to me what the intention is with these 3 checks.

Reply:
@jridgewell 
In order to make this PR easy to merge, I was avoiding any cases that needed special treatment, like fall-through cases etc.. i.e. simple branches.

I was trying to define what a "simple branch" might look like, I came up with this logic:
```js
case 1: // "empty"
case 2:
  i = 200;
  break; // "has a break"
case 3:  
  return 100; // "has a return"
``